src/modules/filelist/cells/SelectBox.jsx

```diff
@@ -6,23 +6,30 @@ import { TableCell } from 'cozy-ui/transpiled/react/deprecated/Table'
 
 import styles from '@/styles/filelist.styl'
 
-const SelectBox = ({ withSelectionCheckbox, selected, onClick, disabled }) => (
-  <TableCell
-    className={cx(
-      styles['fil-content-cell'],
-      styles['fil-content-file-select']
-    )}
-    {...(!disabled && { onClick })}
-  >
-    {withSelectionCheckbox && !disabled && (
-      <Checkbox
-        checked={selected}
-        onChange={() => {
-          // handled by onClick on the <TableCell>
-        }}
-      />
-    )}
-  </TableCell>
-)
+import { useViewSwitcherContext } from '@/lib/ViewSwitcherContext'
+
+const SelectBox = ({ withSelectionCheckbox, selected, onClick, disabled }) => {
```

Comment:
This component should be dump, and should not rely on the hook. Pass a `viewType` props directly.